pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,308 @@ import (
-			})
-
-			Convey("interpolate $__interval in $__timeGroup", func() {
-				sql, err := Interpolate(query, timeRange, "select $__timeGroupAlias(time,$__interval)")
-				So(err, ShouldBeNil)
-
-				So(sql, ShouldEqual, "select $__timeGroupAlias(time,1m)")
-			})
+	dt := time.Date(2018, 3, 14, 21, 20, 6, int(527345*time.Microsecond), time.UTC)
+	earlyDt := time.Date(1970, 3, 14, 21, 20, 6, int(527345*time.Microsecond), time.UTC)
+
+	t.Run("Given a time range between 2018-04-12 00:00 and 2018-04-12 00:05", func(t *testing.T) {
+		from := time.Date(2018, 4, 12, 18, 0, 0, 0, time.UTC)
+		to := from.Add(5 * time.Minute)
+		timeRange := tsdb.NewFakeTimeRange("5m", "now", to)
+		query := &tsdb.Query{DataSource: &models.DataSource{}, Model: simplejson.New()}
+
+		t.Run("interpolate $__interval", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__interval ")
+			require.NoError(t, err)
+			require.Equal(t, "select 1m ", sql)
+		})
 
-			Convey("interpolate $__interval_ms", func() {
-				sql, err := Interpolate(query, timeRange, "select $__interval_ms ")
-				So(err, ShouldBeNil)
+		t.Run("interpolate $__interval in $__timeGroup", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__timeGroupAlias(time,$__interval)")
+			require.NoError(t, err)
+			require.Equal(t, "select $__timeGroupAlias(time,1m)", sql)
+		})
 
-				So(sql, ShouldEqual, "select 60000 ")
-			})
+		t.Run("interpolate $__interval_ms", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__interval_ms ")
+			require.NoError(t, err)
+			require.Equal(t, "select 60000 ", sql)
+		})
 
-			Convey("interpolate __unixEpochFrom function", func() {
-				sql, err := Interpolate(query, timeRange, "select $__unixEpochFrom()")
-				So(err, ShouldBeNil)
+		t.Run("interpolate __unixEpochFrom function", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__unixEpochFrom()")
+			require.NoError(t, err)
+			require.Equal(t, fmt.Sprintf("select %d", from.Unix()), sql)
+		})
 
-				So(sql, ShouldEqual, fmt.Sprintf("select %d", from.Unix()))
-			})
+		t.Run("interpolate __unixEpochTo function", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__unixEpochTo()")
+			require.NoError(t, err)
+			require.Equal(t, fmt.Sprintf("select %d", to.Unix()), sql)
+		})
+	})
 
-			Convey("interpolate __unixEpochTo function", func() {
-				sql, err := Interpolate(query, timeRange, "select $__unixEpochTo()")
-				So(err, ShouldBeNil)
+	t.Run("Given row values with time.Time as time columns", func(t *testing.T) {
+		var nilPointer *time.Time
+
+		fixtures := make([]interface{}, 5)
+		fixtures[0] = dt
+		fixtures[1] = &dt
+		fixtures[2] = earlyDt
+		fixtures[3] = &earlyDt
+		fixtures[4] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
```

Comment:
Do you mind just removing `t.Run` here? Every `t.Run` introduces a sub-test, and I don't think it makes sense to group assertions as sub-tests. IMO a sub-test should correspond to a test case.

Reply:
👍  Was not sure about the wanted format, thanks for clarifying.